src/view.c

```diff
@@ -1285,9 +1342,10 @@ view_snap_to_edge(struct view *view, enum view_edge edge, bool store_natural_geo
 		}
 	}
 
-	if (view->maximized) {
+	if (view->maximized != VIEW_AXIS_NONE) {
```

Comment:
Similar here, e.g. we could just keep using `if (view->maximized)` (to catch any kind of maximize). But again, personal preference.